### PR TITLE
docs: update `bigframes.pandas.Index` docstrings

### PR DESCRIPTION
**Pull request by arwas11**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)
  - `values`: screen/4ViVk2h34LfgD9r
  - `shape`: screen/3Ld4EGsxTfmZnVh
  - `nlevels`: screen/6TYw5qPWkzpKFq7
  - `is_unique`: screen/7grdQKsxVLv6mGs
  - `has_duplicates`: screen/7CQgVfVsiNtkqXw
  - `dtype`: screen/ 75HzCFFRdPbJVCn
  - `dtypes`: screen/6i8zVhpfksEdzVR
  - `T`: screen/9ZuShN6bfoWvKxn
  - `copy`: screen/8pMYrfDSaLqcdQM
  - `astype`: screen/8AsiSd6DvjszHkQ, screen/
  - `get_level_values`: screen/ 
  - `to_series`: screen/, screen/
  - `isin`: screen/3G6UfR5fTocREdn, screen/86tTDdCKnSWdVZx
  - `all`: screen/8w3maL5nEkp9Jko, screen/AT9SB55jdUkwViC
  - `any`: screen/6zbST5graLENBcX
  - `min`: screen/c38rufXA7pdZq22
  - `max`: screen/8RpYJJTBCcgZZ3x
  - `argmin`: screen/7CqEuNjhjibmXJE, screen/5D67hEv7okLXhVG
  - `argmax`: screen/4ojQtUpLqEW2qDE, screen/6i63mdEUgFaVArH
  - `nunique`: screen/ 4zAMoEvxFGLQ2vT
  - `sort_values`: screen/5nsxukvUq5raPHe, screen/AcviuSvz4m5fuTs
  - `value_counts`: screen/AMh3QtVxLVBhPDT, screen/SKjbGAzjEZhAswS, screen,/ByGcJANuAiJkZjK
  - `fillna`: screen/4f7nZJRqBZ3Vyb2
  - `rename`: screen/4zvSA3knkh6stpB
  - `drop`: screen/7dTVqr9DAEtBUsQ
  - `dropna`: screen/5cxdPbQjSq8WHiy
  - `drop_duplicates`: screen/BkKbQXgvZRPfEbN, screen/3ggJqu9RQj6BqFY

Fixes #296390934 🦕

### CODE REVIEW

File: bigframes/core/indexes/base.py [hunk 1]
```diff
@@ -134,7 +134,6 @@ def name(self, value: blocks.Label):
 
     @property
     def names(self) -> typing.Sequence[blocks.Label]:
-        """Returns the names of the Index."""
```
Comment (tswast) [inline bigframes/core/indexes/base.py:137]: For context for Huan, these are borrowed from pandas, so need to come from `third_party`.

**Review by tswast — APPROVED** [review]
Love it! The examples make me happy.

**Comment by tswast** [comment]
Please make sure `doctest` passes before merging.

File: third_party/bigframes_vendored/pandas/core/indexes/base.py [hunk 2]
```diff
@@ -172,44 +396,138 @@ def isin(self, values):
...
+
+            >>> import bigframes.pandas as bpd
+            >>> bpd.options.display.progress_bar = None
+
+            >>> idx = bpd.Index([3, 2, 1])
+            >>> idx.max()
+            3
+
+            >>> idx = bpd.Index(['c', 'b', 'a'])
+            >>>> idx.max()
```
Comment (Genesis929) [inline third_party/bigframes_vendored/pandas/core/indexes/base.py:525]: should be >>>, doctest failed.

**Comment by tswast** [comment]
We're still getting quite a few doctest failures. Please fix.

```
FAILED third_party/bigframes_vendored/pandas/core/indexes/base.py::indexes.base.Index.all
FAILED third_party/bigframes_vendored/pandas/core/indexes/base.py::indexes.base.Index.any
FAILED third_party/bigframes_vendored/pandas/core/indexes/base.py::indexes.base.Index.nlevels
FAILED third_party/bigframes_vendored/pandas/core/indexes/base.py::indexes.base.Index.argmax
FAILED third_party/bigframes_vendored/pandas/core/indexes/base.py::indexes.base.Index.argmin
FAILED third_party/bigframes_vendored/pandas/core/indexes/base.py::indexes.base.Index.nunique
FAILED third_party/bigframes_vendored/pandas/core/indexes/base.py::indexes.base.Index.dropna
FAILED third_party/bigframes_vendored/pandas/core/indexes/base.py::indexes.base.Index.has_duplicates
FAILED third_party/bigframes_vendored/pandas/core/indexes/base.py::indexes.base.Index.isin
FAILED third_party/bigframes_vendored/pandas/core/indexes/base.py::indexes.base.Index.max
FAILED third_party/bigframes_vendored/pandas/core/indexes/base.py::indexes.base.Index.min
FAILED third_party/bigframes_vendored/pandas/core/indexes/base.py::indexes.base.Index.name
```

File: third_party/bigframes_vendored/pandas/core/indexes/base.py [hunk 3]
```diff
@@ -172,44 +396,138 @@ def isin(self, values):
...
 
+        **Examples:**
+
+            >>> import bigframes.pandas as bpd
+            >>> bpd.options.display.progress_bar = None
+
+        True, because nonzero integers are considered True.
+
+            >>> bpd.Index([1, 2, 3]).all()
+            True
```
Comment (tswast) [inline third_party/bigframes_vendored/pandas/core/indexes/base.py:447]: Unfortunately, `pandas` gives a `numpy` object in some versions. It should be interchangeable with a bool, but it has a different repr. Try this (here and in other failing tests):

```suggestion
            >>> bool(bpd.Index([1, 2, 3]).all())
            True
```

Comment (arwas11) [reply third_party/bigframes_vendored/pandas/core/indexes/base.py:447]: Thanks for the suggestion, I'll give it a try.

**Pull request merged**